cluster/juju/layers/kubernetes-master/build/update-addons

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/python3
+
+description = """
+Update addon manifests for the charm.
+
+This will clone the kubernetes repo and place the addons in
+<charm>/templates/addons.
+
+Can be run with no arguments and from any folder.
+"""
+
+import argparse
+import os
+import shutil
+import subprocess
+
+os.chdir(os.path.dirname(__file__))
+target = os.path.abspath("../templates/addons")
+
+def sh(cmd):
+    """ Alias for subprocess calls """
+    subprocess.check_call(cmd.split())
+
+def clean_addons_folder():
+    """ Remove and recreate the addons folder """
+    print("Cleaning addons folder")
+    shutil.rmtree(target, ignore_errors=True)
+    os.mkdir(target)
+
+def get_kubernetes_repo():
+    """ Git clone the kubernetes repo. If the repo already exists from a
+    previous build, git pull instead. """
+    print("Getting the Kubernetes repo")
+    if os.path.exists("kubernetes"):
+        os.chdir("kubernetes")
+        sh("git pull")
+        os.chdir("..")
+    else:
+        sh("git clone https://github.com/kubernetes/kubernetes.git")
+
+def add_addon(source, name=None):
+    """ Add an addon manifest from the given source.
+
+    Any occurrences of 'amd64' are replaced with '{{ arch }}' so the charm can
+    fill it in during deployment. """
+    name = name or os.path.basename(source)
+    dest = target + "/" + name
+    print("'%s' -> '%s'" % (source, dest))
+    with open(source, "r") as f:
+        content = f.read()
+    content = content.replace("amd64", "{{ arch }}")
+    with open(target + "/" + name, "w") as f:
+        f.write(content)
+
+def update_addons():
+    """ Update addons. This will clean the addons folder and add new manifests
+    from upstream. """
+    clean_addons_folder()
+    get_kubernetes_repo()
+    print("Adding addons")
+    add_addon("kubernetes/cluster/addons/dashboard/dashboard-controller.yaml")
+    add_addon("kubernetes/cluster/addons/dashboard/dashboard-service.yaml")
+    add_addon("kubernetes/cluster/addons/dns/skydns-rc.yaml.in", "kubedns-rc.yaml")
+    add_addon("kubernetes/cluster/addons/dns/skydns-svc.yaml.in", "kubedns-svc.yaml")
+    add_addon("kubernetes/cluster/addons/cluster-monitoring/influxdb/grafana-service.yaml")
+    add_addon("kubernetes/cluster/addons/cluster-monitoring/influxdb/heapster-controller.yaml")
+    add_addon("kubernetes/cluster/addons/cluster-monitoring/influxdb/heapster-service.yaml")
+    add_addon("kubernetes/cluster/addons/cluster-monitoring/influxdb/influxdb-grafana-controller.yaml")
+    add_addon("kubernetes/cluster/addons/cluster-monitoring/influxdb/influxdb-service.yaml")
+
+def parse_args():
+    parser = argparse.ArgumentParser(description=description)
+    parser.parse_args()
+
+if __name__ == "__main__":
+    parse_args()
+    update_addons()
```

Comment:
It could. The reason it landed this way is 'cause I wrote and tested in python3, but the tactic is python2, and I didn't want to be tied to that. Doesn't really make any difference to me though.